packages/components/src/ClipboardButton/ClipboardButton.js

```diff
@@ -0,0 +1,37 @@
+import { contextConnect, useContextSystem } from '@wp-g2/context';
+import { useClipboard } from '@wp-g2/utils';
+import { noop } from 'lodash';
+import React from 'react';
+import { useEffect } from 'react';
+import { useRef } from 'react';
+
+import { Button } from '../Button';
+
+function ClipboardButton(props, forwardedRef) {
+	const {
+		text,
+		onCopy: onCopyProp = noop,
+		onFinishCopy = noop,
+		...otherProps
+	} = useContextSystem(props, 'ClipboardButton');
+	const { hasCopied, onCopy } = useClipboard(text);
+	const lastHasCopied = useRef(hasCopied);
+
+	useEffect(() => {
+		if (lastHasCopied.current === hasCopied) {
+			return;
+		}
+
+		if (hasCopied) {
+			onCopyProp();
+		} else {
+			onFinishCopy();
+		}
+
+		lastHasCopied.current = hasCopied;
+	}, [hasCopied, onCopyProp, onFinishCopy]);
+
+	return <Button {...otherProps} onClick={onCopy} ref={forwardedRef} />;
```

Comment:
@saramarcondes Would love your thoughts on this. I think we should pass-through an `onClick` callback as well (from props), in addition to firing the `onCopy` callback. I'd expect `onClick` to work (as a consumer), even though under the hood it's firing `onCopy`. What do you think :)

Reply:
I had the same thought, I'm glad you brought it up. I think it makes sense to merge the two as you suggested 👍